### PR TITLE
Document options managed automatically

### PR DESCRIPTION
**Pull request by danimtb**
Docs for https://github.com/conan-io/conan/pull/14194

### CODE REVIEW

File: reference/tools/cpp_info.rst [hunk 1]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+.. code-block:: python
+
+    from conan.tools import default_configure
+
+    def configure(self):
+        default_configure(self)
+
+.. _conan_tools_default_package_id:
+
+conan.tools.default_pacakge_id
```
Comment (czoido) [inline reference/tools/cpp_info.rst:128]:
```suggestion
conan.tools.default_package_id
```

File: reference/conanfile/methods/config_options.rst [hunk 2]
```diff
@@ -3,17 +3,48 @@
 config_options()
 ================
 
-The ``config_options()`` method
-is used to configure or constraint the available options in a package, **before** they are given a value. A typical use case is to remove an option in a given platform. For example,
-the ``fPIC`` flag doesn't exist in Windows, so it should be removed in this method like so:
+The ``config_options()`` method is used to configure or constraint the available options in a package, **before** they are given a value.
+A typical use case is to remove an option in a given platform.
+For example, the ``SSE2`` flag doesn't exist in architectures differnt than 32 bits, so it should be removed in this method like so:
```
Comment (czoido) [inline reference/conanfile/methods/config_options.rst:8]:
```suggestion
For example, the ``SSE2`` flag doesn't exist in architectures different than 32 bits, so it should be removed in this method like so:
```

File: reference/tools/cpp_info.rst [hunk 3]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+        if conanfile.options.get_safe("header_only") or conanfile.package_type is "header-library":
+            conanfile.info.clear()
+
+Usage:
+
+.. code-block:: python
+
+    from conan.tools import default_configure
+
+    def pacakge_id(self):
```
Comment (czoido) [inline reference/tools/cpp_info.rst:155]:
```suggestion
    def package_id(self):
```

File: reference/conanfile/methods/package_id.rst [hunk 4]
```diff
@@ -12,6 +12,33 @@ The general rule is that every different value of ``settings`` and ``options`` c
 - A given package recipe can implement some partial erasure of information, for example to obtain the same ``package_id`` for a range of compiler versions. This type of binary compatibility is in general better addressed with the global ``compatibility`` plugin, or with the ``compatibility()`` method if the global plugin is not enough.
 - A package recipe can decide to inject extra variability in its computed ``package_id``, adding ``conf`` items or "target" settings.
 
+Default behavior
+++++++++++++++++
+
+.. include:: ../../../common/experimental_warning.inc
+
+When ``pacakge_id()`` method is not defined, Conan manages the package ID automatically clearing settings and options when the recipe
```
Comment (czoido) [inline reference/conanfile/methods/package_id.rst:20]:
```suggestion
When ``package_id()`` method is not defined, Conan manages the package ID automatically clearing settings and options when the recipe
```

File: reference/conanfile/methods/config_options.rst [hunk 5]
```diff
@@ -3,17 +3,48 @@
 config_options()
 ================
 
-The ``config_options()`` method
-is used to configure or constraint the available options in a package, **before** they are given a value. A typical use case is to remove an option in a given platform. For example,
-the ``fPIC`` flag doesn't exist in Windows, so it should be removed in this method like so:
+The ``config_options()`` method is used to configure or constraint the available options in a package, **before** they are given a value.
```
Comment (franramirez688) [inline reference/conanfile/methods/config_options.rst:6]:
```suggestion
The ``config_options()`` method is used to configure or constrain the available options in a package **before** assigning them a value.
```

File: reference/conanfile/methods/config_options.rst [hunk 6]
```diff
@@ -3,17 +3,48 @@
...
 
     def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
+        if self.settings.arch != "x86_64":
+            del self.options.with_sse2
 
-The ``config_options()`` method executes before the ``configure()`` method, and before the actual assignment of the ``options`` values, but after settings are already defined.
+The ``config_options()`` method executes before the ``configure()`` method, and before the actual assignment of the ``options`` values,
+but after settings are already defined.
```
Comment (franramirez688) [inline reference/conanfile/methods/config_options.rst:17]: I think we could rephrase it to make it easier to read:

```
The ``config_options()`` method executes:
* Before calling the ``configure()`` method.
* Before assigning the ``options`` values.
* After ``settings`` are already defined.
```
WDYT?

File: reference/conanfile/methods/config_options.rst [hunk 7]
```diff
@@ -3,17 +3,48 @@
...
+- ``fPIC`` (True, False).
+
+To opt-out from this behavior, the method can be empty-defined:
+
+.. code-block:: python
+
+    def config_options(self):
+        pass
+
+To manage extra options appart from the ones automatically managed, the tool has to be called explicitly:
```
Comment (franramirez688) [inline reference/conanfile/methods/config_options.rst:38]:
```suggestion
To manage extra options apart from the ones automatically handled, the tool has to be explicitly called:
```

File: reference/conanfile/methods/configure.rst [hunk 8]
```diff
@@ -43,6 +43,38 @@ so it should be removed:
...
+- ``header_only`` (True, False).
+
+To opt-out from this behavior, the method can be empty-defined:
+
+.. code-block:: python
+
+    def configure(self):
+        pass
+
+To manage extra options appart from the ones automatically managed, the tool has to be called explicitly:
```
Comment (franramirez688) [inline reference/conanfile/methods/configure.rst:67]:
```suggestion
To manage extra options apart from the ones automatically handled, the tool has to be explicitly called:
```

File: reference/conanfile/methods/package_id.rst [hunk 9]
```diff
@@ -12,6 +12,33 @@ The general rule is that every different value of ``settings`` and ``options`` c
 - A given package recipe can implement some partial erasure of information, for example to obtain the same ``package_id`` for a range of compiler versions. This type of binary compatibility is in general better addressed with the global ``compatibility`` plugin, or with the ``compatibility()`` method if the global plugin is not enough.
 - A package recipe can decide to inject extra variability in its computed ``package_id``, adding ``conf`` items or "target" settings.
 
+Default behavior
+++++++++++++++++
+
+.. include:: ../../../common/experimental_warning.inc
+
+When ``package_id()`` method is not defined, Conan manages the package ID automatically clearing settings and options when the recipe
```
Comment (franramirez688) [inline reference/conanfile/methods/package_id.rst:20]:
```suggestion
When the ``package_id()`` method is not defined, Conan automatically manages the package ID clearing settings and options when the recipe
```

File: reference/conanfile/methods/package_id.rst [hunk 10]
```diff
@@ -12,6 +12,33 @@ The general rule is that every different value of ``settings`` and ``options`` c
...
+This is achived by calling the :ref:`conan.tools.default_package_id()<conan_tools_default_package_id>` tool.
+
+To opt-out from this behavior, the method can be empty-defined:
+
+.. code-block:: python
+
+    def package_id(self):
+        pass
+
+To manage the package ID info further, the tool has to be called explicitly:
```
Comment (franramirez688) [inline reference/conanfile/methods/package_id.rst:31]:
```suggestion
To manage the package ID info further, the tool has to be explicitly called:
```

File: reference/tools/cpp_info.rst [hunk 11]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+.. _conan_tools_default_config_options:
+
+conan.tools.default_config_options
+==================================
+
+.. include:: ../../common/experimental_warning.inc
+
+.. note::
+
+    This tool is called automatically **only** whenever a recipe does not declare a ``config_options()`` method explicitly.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:61]:
```suggestion
    This tool is automatically called **only** whenever a recipe does not declare a ``config_options()`` method explicitly.
```

File: reference/tools/cpp_info.rst [hunk 12]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+
+Manage options in `config_options()` method automatically. This tool manages the following options:
+
+- ``fPIC`` (True, False): Option set as convention to designate "Position Independent Code" flag.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:65]:
```suggestion
- ``fPIC`` (True, False): Option set as a convention to designate "Position Independent Code" flag.
```

File: reference/tools/cpp_info.rst [hunk 13]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+.. _conan_tools_default_configure:
+
+conan.tools.default_configure
+=============================
+
+.. include:: ../../common/experimental_warning.inc
+
+.. note::
+
+    This tool is called automatically **only** whenever a recipe does not declare a ``configure()`` method explicitly.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:97]:
```suggestion
    This tool is automatically called **only** whenever a recipe does not declare a ``configure()`` method explicitly.
```

File: reference/tools/cpp_info.rst [hunk 14]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+
+Manage options in `configure()` method automatically. This tool manages the following options:
+
+- ``fPIC`` (True, False): Option set as convention to designate "Position Independent Code" flag.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:101]:
```suggestion
- ``fPIC`` (True, False): Option set as a convention to designate "Position Independent Code" flag.
```

File: reference/tools/cpp_info.rst [hunk 15]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+  This option is removed when there is a ``header_only=True`` option or when ``shared=True``.
+- ``shared`` (True, False): Option set as convention to designate a dynamic library.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:103]:
```suggestion
- ``shared`` (True, False): Option set as a convention to designate a dynamic library.
```

File: reference/tools/cpp_info.rst [hunk 16]
```diff
@@ -48,3 +48,109 @@ The public documented interface (besides the defined one in :ref:`the package_in
...
+.. _conan_tools_default_package_id:
+
+conan.tools.default_package_id
+==============================
+
+.. include:: ../../common/experimental_warning.inc
+
+.. note::
+
+    This tool is called automatically **only** whenever a recipe does not declare a ``package_id()`` method explicitly.
```
Comment (franramirez688) [inline reference/tools/cpp_info.rst:135]:
```suggestion
    This tool is automatically called **only** whenever a recipe does not declare a ``package_id()`` method explicitly.
```

**Review by franramirez688 — COMMENTED** [review]
Great explanation @danimtb ! 👏  Only a few wording comments (we can discuss them for sure).

File: reference/conanfile/methods/config_options.rst (hunk 6 above)
Comment (danimtb) [reply reference/conanfile/methods/config_options.rst:17]: I did not change the wording around this, just split the line to make it easier to read in the editor. Anyway, I can add your suggestion

Comment (danimtb) [reply reference/conanfile/methods/config_options.rst]: done

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]
